src/base/emu-i386/simx86/memory.c

```diff
@@ -433,8 +433,10 @@ int e_handle_pagefault(sigcontext_t *scp)
 	 * linked by Cpatch will do it */
 	/* ACH: we can set up a data patch for code
 	 * which has not yet been executed! */
+#ifdef SKIP_CPATCH
```

Comment:
Wow!
Is this correct?
I suspect these patches were done
in a big hurry.

Reply:
This should have been #ifndef obviously. What can I say, I had a beer.

On the positive side this exposed some bugs in the JIT when using MSDOS. I have now found what caused them, will file PRs tomorrow (https://gist.github.com/fa0de21b11ddeb750441d7f0dcc93cb3 is the idea here)